regression/apps/.clang-format

```diff
@@ -0,0 +1,125 @@
+# SPDX-License-Identifier: GPL-2.0
+#
+# This file was copied from Linux kernel, or more precisely, from:
+#
+#   https://github.com/torvalds/linux/blob/e5eb28f6d1afebed4bb7d740a797d0390bd3a357/.clang-format
+#
+# We have removed the long but useless `ForEachMacros` option.
+
+# clang-format configuration file. Intended for clang-format >= 11.
+#
+# For more information, see:
+#
+#   Documentation/process/clang-format.rst
+#   https://clang.llvm.org/docs/ClangFormat.html
+#   https://clang.llvm.org/docs/ClangFormatStyleOptions.html
+#
+---
+AccessModifierOffset: -4
+AlignAfterOpenBracket: Align
+AlignConsecutiveAssignments: false
+AlignConsecutiveDeclarations: false
+AlignEscapedNewlines: Left
+AlignOperands: true
+AlignTrailingComments: false
+AllowAllParametersOfDeclarationOnNextLine: false
+AllowShortBlocksOnASingleLine: false
+AllowShortCaseLabelsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: None
+AllowShortIfStatementsOnASingleLine: false
+AllowShortLoopsOnASingleLine: false
+AlwaysBreakAfterDefinitionReturnType: None
+AlwaysBreakAfterReturnType: None
+AlwaysBreakBeforeMultilineStrings: false
+AlwaysBreakTemplateDeclarations: false
+BinPackArguments: true
+BinPackParameters: true
+BraceWrapping:
+  AfterClass: false
+  AfterControlStatement: false
+  AfterEnum: false
+  AfterFunction: true
+  AfterNamespace: true
+  AfterObjCDeclaration: false
+  AfterStruct: false
+  AfterUnion: false
+  AfterExternBlock: false
+  BeforeCatch: false
+  BeforeElse: false
+  IndentBraces: false
+  SplitEmptyFunction: true
+  SplitEmptyRecord: true
+  SplitEmptyNamespace: true
+BreakBeforeBinaryOperators: None
+BreakBeforeBraces: Custom
+BreakBeforeInheritanceComma: false
+BreakBeforeTernaryOperators: false
+BreakConstructorInitializersBeforeComma: false
+BreakConstructorInitializers: BeforeComma
+BreakAfterJavaFieldAnnotations: false
+BreakStringLiterals: false
+ColumnLimit: 80
+CommentPragmas: '^ IWYU pragma:'
+CompactNamespaces: false
+ConstructorInitializerAllOnOneLineOrOnePerLine: false
+ConstructorInitializerIndentWidth: 8
+ContinuationIndentWidth: 8
+Cpp11BracedListStyle: false
+DerivePointerAlignment: false
+DisableFormat: false
+ExperimentalAutoDetectBinPacking: false
+FixNamespaceComments: false
+
+IncludeBlocks: Preserve
+IncludeCategories:
+  - Regex: '.*'
+    Priority: 1
+IncludeIsMainRegex: '(Test)?$'
+IndentCaseLabels: false
+IndentGotoLabels: false
+IndentPPDirectives: None
+IndentWidth: 8
```

Comment:
Maybe Linux is too stubborn using a tab width of 8 and using tabs instead of spaces.

Nevertheless, the debates on this topic will never come to consensus. But I suggest Asterinas to follow the least aggressive coding style using 4 spaces for tabs.

Reply:
Personally, I am fine with the unmodified code style of the Linux kernel. There are two reasons for this:
 - Using 8 spaces as indentation may not be appropriate for "advanced" languages such as Rust and Java, because such languages have "advanced" features that require more levels of indentation. For example, there are streams and closures, so whenever we pass a closure as an argument (e.g., to `.iter().map()`), we need another level of indentation (two levels of indentation if methods are chained in multiple lines). Besides, `impl` in Rust and `class` in Java also contribute another level of indentation. In C, however, all of the above arguments do not hold, so I agree with [the Linux kernel documentation's / Linux Torvald's argument for 8-space indentation](https://github.com/torvalds/linux/blob/66a27abac311a30edbbb65fe8c41ff1b13876faa/Documentation/process/coding-style.rst?plain=1#L31-L35):
> Now, some people will claim that having 8-character indentations makes the code move too far to the right, and makes it hard to read on a 80-character terminal screen. The answer to that is that if you need more than 3 levels of indentation, you're screwed anyway, and should fix your program.
 - I would like to avoid unnecessary changes to this format file. Changing the way code is indented seems like a very big change, and after we make such changes, I do not think we can call such formatting rules "Linux kernel code style" anymore. Linux kernel code style clearly requires us to use 8-space indentation, with no other alternatives.

Reply:
I am ok with 8-space tabs. Using wider tabs would make deeply nested code intolerably ugly, which compels people to refactor the code.